src/driver/src/main/java/com/edgedb/driver/EdgeDBConnection.java

```diff
@@ -464,7 +464,14 @@ public static EdgeDBConnection fromProjectFile(@NotNull File file) throws IOExce
             throw new FileNotFoundException("Could not find instance name under project directory " + projectDir);
         }
 
-        return fromInstanceName(instanceDetails.get().getLinkedInstanceName(), instanceDetails.get().getProfile());
+        var connection = fromInstanceName(instanceDetails.get().getLinkedInstanceName(), instanceDetails.get().getProfile());
+
+        Optional<String> database;
+        if((database = ConfigUtils.tryResolveProjectDatabase(projectDir)).isPresent()) {
```

Comment:
This is more of a question about Java idioms, but why have the assignment within the if statement?

```suggestion
        Optional<String> database = ConfigUtils.tryResolveProjectDatabase(projectDir);
        if(database.isPresent()) {
```

seems a bit more clear?

Reply:
Yeah that makes things more readable, I forget why I in-lined it lol